packages/engine-server/src/markdown/remark/utils.ts

```diff
@@ -216,6 +219,47 @@ export class LinkUtils {
   }
 }
 
+export class AnchorUtils {
+  static async findAnchors(opts: {
+    note: NoteProps;
+    wsRoot: string;
+  }): Promise<{ [index: string]: DNoteAnchorPositioned }> {
+    if (opts.note.stub) return {};
+    const noteContents = await NoteUtils.readFullNote(opts);
+    const noteAnchors = RemarkUtils.findAnchors(noteContents);
+    const anchors: [string, DNoteAnchorPositioned][] = [];
+    noteAnchors.forEach((anchor) => {
+      if (_.isUndefined(anchor.position)) return;
+      const { line, column } = anchor.position.start;
+      if (anchor.type === "heading") {
+        const value = getSlugger().slug(anchor.children[0].value as string);
```

Comment:
we should instantiate `getSlugger` at the top. github slugger will add the correct suffix to the header if it encounters duplicates